### PR TITLE
minor js fixes & name chunks

### PR DESCRIPTION
**Pull request by davidejones**
### What does this PR do?

This pr includes:
- some minor js bug fixes from error-tracking
- specifically naming chunks (previous to this vendor was just output as 0.js)

### Motivation

Debugging failed release

### Preview

Check that browser the nav and async loading is working as normal
Check the integrations / security rules listings make sure thats working as normal
https://docs-staging.datadoghq.com/david.jones/js-fixes/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: src/scripts/components/async-loading.js [hunk 1]
```diff
@@ -205,7 +205,7 @@ function loadPage(newUrl) {
             gtag('config', gaTag, { page_path: pathName });
 
             // Marketo
-            if (typeof window.Munchkin !== 'undefined') {
+            if (typeof window.Munchkin !== 'undefined' && typeof window.Munchkin.munchkinFunction === 'function') {
```
Comment (davidejones) [inline src/scripts/components/async-loading.js:208]: Note: fixing for error `window.Munchkin.munchkinFunction is not a function`

File: src/scripts/components/security-rules.js [hunk 2]
```diff
@@ -23,7 +23,8 @@ export function initializeSecurityRules() {
         let results = [];
 
         if (filterCategoryValue && filterCategoryValue !== 'all') {
-            results = document.querySelectorAll(`.${filterCategoryValue}`);
+            const val = (filterCategoryValue.startsWith('.')) ? filterCategoryValue.slice(1) : filterCategoryValue;
+            results = document.querySelectorAll(decodeURI(`.${val}`));
```
Comment (davidejones) [inline src/scripts/components/security-rules.js:27]: Note: fixing for error `Failed to execute 'querySelectorAll' on 'Document': '..cat-aws' is not a valid selector.`

File: src/scripts/components/security-rules.js [hunk 3]
```diff
@@ -54,10 +55,13 @@ export function initializeSecurityRules() {
 
     const handleEmptyResultSet = () => {
         const searchQuery = inputSearch.value;
-        const activeCategoryFilter = stringToTitleCase(document.querySelector('.controls .active').text);
+        const txt = (document.querySelector('.controls .active')) ? document.querySelector('.controls .active').text : '';
+        const activeCategoryFilter = stringToTitleCase(txt);
```
Comment (davidejones) [inline src/scripts/components/security-rules.js:59]: Note: fixing for error `null is not an object (evaluating 'document.querySelector(".controls .active").text')`

File: src/scripts/components/security-rules.js [hunk 4]
```diff
@@ -54,10 +55,13 @@ export function initializeSecurityRules() {
...
         const message = `No results found for query "${searchQuery}" in category ${activeCategoryFilter}`;
-        jsEmptyResults.innerText = message;
-        jsEmptyResults.classList.remove('d-none');
+        if(jsEmptyResults) {
+          jsEmptyResults.innerText = message;
+          jsEmptyResults.classList.remove('d-none');
+        }
```
Comment (davidejones) [inline src/scripts/components/security-rules.js:64]: Note: fixing for error `Cannot set property 'innerText' of null`

File: src/scripts/components/security-rules.js [hunk 5]
```diff
@@ -77,7 +81,7 @@ export function initializeSecurityRules() {
         // Handle empty result set
         if (filteredResults.length < 1) {
             handleEmptyResultSet();
-        } else {
+        } else if (jsEmptyResults) {
```
Comment (davidejones) [inline src/scripts/components/security-rules.js:84]: Note: fixing for error Cannot set property 'innerText' of null

File: src/scripts/components/security-rules.js [hunk 6]
```diff
@@ -105,7 +109,8 @@ export function initializeSecurityRules() {
 
     const handleKeyup = () => {
         const searchValue = inputSearch.value.length > 2 ? inputSearch.value.toLowerCase().trim() : '';
-        const activeCategoryFilter = document.querySelector('.controls .active').dataset.filter;
+        const activeCategory = document.querySelector('.controls .active');
+        const activeCategoryFilter = (activeCategory) ? activeCategory.dataset.filter : '';
```
Comment (davidejones) [inline src/scripts/components/security-rules.js:113]: Note: fixing for document.querySelector(".controls .active") is null

**Review by bgdeutsch — APPROVED** [review]
LGTM!  Tested security rules and was unable to produce any JS console errors.   Checking sources `0.js` is no longer showing and i'm seeing `vendor.js` with the chunkhash appended.

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**